hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/volume/DeleteVolumeHandler.java

```diff
@@ -92,16 +85,17 @@ protected void execute(OzoneClient client, OzoneAddress address)
     String volumeName = address.getVolumeName();
     try {
       if (bRecursive) {
-        Collection<String> serviceIds = getConf().getTrimmedStringCollection(
-            OZONE_OM_SERVICE_IDS_KEY);
-        if (Strings.isNullOrEmpty(omServiceId)) {
-          if (serviceIds.size() > 1) {
-            out().printf("OmServiceID not provided, provide using " +
-                "-id <OM_SERVICE_ID>%n");
-            return;
-          } else if (serviceIds.size() == 1) {
+        if (!Strings.isNullOrEmpty(address.getOmHost())) {
+          omServiceId = address.getOmHost();
+        } else {
+          Collection<String> serviceIds = getConf().getTrimmedStringCollection(
+              OZONE_OM_SERVICE_IDS_KEY);
+          if (serviceIds.size() == 1) {
             // Only one OM service ID configured, we can use that
+            // If more than 1, it will fail in createClient step itself
             omServiceId = serviceIds.iterator().next();
+          } else {
+            omServiceId = getConf().get(OZONE_OM_ADDRESS_KEY);
           }
         }
```

Comment:
This is same logic as in `DeleteBucketHandler`.  Can you please extract and reuse?

Reply:
Thanks, extracted in common place and reused.